pkg/controllers/traffictopology/adapter.go

```diff
@@ -0,0 +1,532 @@
+
+func (t *TPControllerAdapter) GetControllerName() string {
+	return ControllerName
+}
+
+func (t *TPControllerAdapter) GetSelectedEmployeeNames(ctx context.Context, employer client.Object) ([]string, error) {
+	return nil, nil
+}
+
+func (t *TPControllerAdapter) GetExpectedEmployer(ctx context.Context, employer client.Object) ([]rsFrameController.IEmployer, error) {
+	var expected []rsFrameController.IEmployer
+
+	trafficTopology, ok := employer.(*v1alpha1.TrafficTopology)
+	if !ok {
+		return expected, fmt.Errorf("not type of TrafficTopology")
+	}
+
+	backendApiVersion := ""
+	backendKind := "Service"
+	if trafficTopology.Spec.Backend.APIVersion != nil {
+		backendApiVersion = *trafficTopology.Spec.Backend.APIVersion
+	}
+	if trafficTopology.Spec.Backend.Kind != nil {
+		backendKind = *trafficTopology.Spec.Backend.Kind
+	}
+	// caution:
+	// for InClusterTrafficType, clusterName of brBackend will be changed
+	brBackend := v1alpha1.CrossClusterObjectReference{
+		ObjectTypeRef: v1alpha1.ObjectTypeRef{
+			APIVersion: backendApiVersion,
+			Kind:       backendKind,
+		},
+		CrossClusterObjectNameReference: v1alpha1.CrossClusterObjectNameReference{
+			Name: trafficTopology.Spec.Backend.Name,
+		},
+	}
+
+	brRoutes := make([]v1alpha1.CrossClusterObjectReference, len(trafficTopology.Spec.Routes))
+	for i, route := range trafficTopology.Spec.Routes {
+		routeApiVersion := "gateway.networking.k8s.io/v1"
+		routeKind := "HTTPRoute"
+		if route.APIVersion != nil {
+			routeApiVersion = *route.APIVersion
+		}
+		if route.Kind != nil {
+			routeKind = *route.Kind
+		}
+		brRoutes[i] = v1alpha1.CrossClusterObjectReference{
+			ObjectTypeRef: v1alpha1.ObjectTypeRef{
+				APIVersion: routeApiVersion,
+				Kind:       routeKind,
+			},
+			CrossClusterObjectNameReference: v1alpha1.CrossClusterObjectNameReference{
+				Name: route.Name,
+			},
+		}
+	}
+
+	switch trafficTopology.Spec.TrafficType {
+	case v1alpha1.MultiClusterTrafficType:
+		br := v1alpha1.BackendRouting{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      employer.GetName() + "-msc",
+				Namespace: employer.GetNamespace(),
+			},
+			Spec: v1alpha1.BackendRoutingSpec{
+				TrafficType: v1alpha1.MultiClusterTrafficType,
+				Backend:     brBackend,
+				Routes:      brRoutes,
+			},
+		}
+		backendRoutingEmployer := TPEmployer{
+			BackendRoutingName: employer.GetName() + "-msc",
+			BackendRouting:     br,
+		}
+
+		workloadInfos := trafficTopology.Spec.WorkloadRef.Match.Names
+		if trafficTopology.Spec.WorkloadRef.Match.Selector != nil {
+			workloads := &unstructured.UnstructuredList{}
+			workloads.SetAPIVersion(trafficTopology.Spec.WorkloadRef.APIVersion)
+			workloads.SetKind(trafficTopology.Spec.WorkloadRef.Kind)
+
+			selector, err := metav1.LabelSelectorAsSelector(trafficTopology.Spec.WorkloadRef.Match.Selector)
+			if err != nil {
+				return expected, err
+			}
+			err = t.List(clusterinfo.WithCluster(ctx, clusterinfo.Clusters), workloads, &client.ListOptions{
```

Comment:
please refer to rollout and rolloutrun controller, you can list workload by workloadregistry

for example
```
	gvk := schema.FromAPIVersionAndKind(obj.Spec.TargetType.APIVersion, obj.Spec.TargetType.Kind)
	store, err := r.workloadRegistry.Get(gvk)
	if err != nil {
		return nil, err
	}
	list, err := store.List(ctx, obj.GetNamespace(), match)
	if err != nil {
		return nil, err
	}
```